python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
+        Installs the requested core and updates core_api.yml.
+        """
+        self._install_core()
+        self._update_core_api_descriptor()
+
+    def _install_core(self):
+        """
         Performs the actual installation of the new version of the core API
         """
-        
-        # validation
-        if self.get_update_status() != TankCoreUpdater.UPDATE_POSSIBLE:
-            raise Exception("Update not allowed at this point. Run get_update_status for details.")
-        
-        # download attachment
-        if self._update_version[constants.TANK_CODE_PAYLOAD_FIELD] is None:
-            raise Exception("Cannot find a binary bundle for %s. Please contact support" % self._update_version["code"])
-        
-        self._log.info("Begin downloading Toolkit Core API %s from the App Store..." % self._update_version["code"])
-        
-        zip_tmp = os.path.join(tempfile.gettempdir(), "%s_tank_core.zip" % uuid.uuid4().hex)
-        extract_tmp = os.path.join(tempfile.gettempdir(), "%s_tank_unzip" % uuid.uuid4().hex)
-        
-        # now have to get the attachment id from the data we obtained. This is a bit hacky.
-        # data example for the payload field, as returned by the query above: 
-        # {'url': 'http://tank.shotgunstudio.com/file_serve/attachment/21', 'name': 'tank_core.zip', 
-        #  'content_type': 'application/zip', 'link_type': 'upload'}
-        # 
-        # grab the attachment id off the url field and pass that to the download_attachment()
-        # method below.
-        
-        try:
-            attachment_id = int(self._update_version[constants.TANK_CODE_PAYLOAD_FIELD]["url"].split("/")[-1])
-        except:
-            raise Exception("Could not extract attachment id from data %s" %  self._update_version)
-        
-        bundle_content = self._sg.download_attachment(attachment_id)
-        fh = open(zip_tmp, "wb")
-        fh.write(bundle_content)
-        fh.close()
-        
-        self._log.info("Download complete - now extracting content...")
-        # unzip core zip file to temp location and run updater
-        unzip_file(zip_tmp, extract_tmp)
-        
-        # and write a custom event to the shotgun event log
-        data = {}
-        data["description"] = "%s: Core API was downloaded" % self._local_sg.base_url
-        data["event_type"] = "TankAppStore_CoreApi_Download"
-        data["entity"] = self._update_version
-        data["user"] = self._sg_script_user
-        data["project"] = constants.TANK_APP_STORE_DUMMY_PROJECT
-        data["attribute_name"] = constants.TANK_CODE_PAYLOAD_FIELD
-        self._sg.create("EventLogEntry", data)
-        
-        
-        self._log.info("Extraction complete - now installing Toolkit Core")
-        sys.path.insert(0, extract_tmp)
+        if not self._new_core_descriptor.exists_local():
+            self._log.info("Begin downloading Toolkit Core API %s from the App Store..." % self._new_core_descriptor.get_version())
```

Comment:
It has, but since we're going to cache info.yml in Shotgun in the future I figured I might leave it in. I'm okay with removing it in the meantime if you'd rather remove it.